### PR TITLE
fixed mnist bug

### PR DESCRIPTION
**Pull request by XiaocenDong**
**Description**: fixed mnist bug, add multi-gpu support

### CODE REVIEW

File: orttraining/orttraining/python/ort_trainer.py [hunk 1]
```diff
@@ -316,8 +316,7 @@ def convert_model_loss_fn_to_onnx(model, loss_fn, model_desc, device, inputs):
                        training=True,
                        _retain_param_name=True,
                        example_outputs=tuple(sample_outputs),
-                       do_constant_folding=False,
-                       **other_export_options)
```
Comment (pengwa) [inline orttraining/orttraining/python/ort_trainer.py:320]: why remove this?

File: orttraining/pytorch_frontend_examples/mnist_training.py [hunk 2]
```diff
@@ -125,7 +128,7 @@ def main():
                         help='input batch size for training (default: 64)')
     parser.add_argument('--test-batch-size', type=int, default=1000, metavar='N',
                         help='input batch size for testing (default: 1000)')
-    parser.add_argument('--epochs', type=int, default=10, metavar='N',
```
Comment (pengwa) [inline orttraining/pytorch_frontend_examples/mnist_training.py:128]: why change this?

File: orttraining/pytorch_frontend_examples/mnist_training.py [hunk 3]
```diff
@@ -175,14 +189,19 @@ def main():
...
     if args.use_ort_trainer:
         # use log_interval as gradient accumulate steps
-        trainer = ORTTrainer(model, my_loss, model_desc, "SGDOptimizer", None, IODescription('Learning_Rate', [1,], torch.float32), device)
+        print(f'world_rank = {args.world_rank}')
+        trainer = ORTTrainer(model, my_loss, model_desc, "LambOptimizer", None, IODescription('Learning_Rate', [1,], torch.float32), device, 1, None,
+        args.world_rank, args.world_size, use_mixed_precision=False, allreduce_post_accumulation = True)
+        print('\nBuild ort model done.')
 
         for epoch in range(1, args.epochs + 1):
+            print("training now\n\n")
```
Comment (pengwa) [inline orttraining/pytorch_frontend_examples/mnist_training.py:198]: Training

File: orttraining/pytorch_frontend_examples/mnist_training.py [hunk 4]
```diff
@@ -167,6 +168,19 @@ def main():
             transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))])),
         batch_size=args.test_batch_size, shuffle=True, **kwargs)
 
+
+    comm = MPI.COMM_WORLD
```
Comment (pengwa) [inline orttraining/pytorch_frontend_examples/mnist_training.py:172]: seems we are assuming this script only work for MPI distributed run, right?

File: orttraining/pytorch_frontend_examples/mnist_training.py [hunk 5]
```diff
@@ -141,19 +144,17 @@ def main():
     parser.add_argument('--save-model', action='store_true', default=False,
                         help='For Saving the current Model')
 
-    parser.add_argument('--use-ort', action='store_true', default=False,
```
Comment (zhijxu-MS) [inline orttraining/pytorch_frontend_examples/mnist_training.py:144]: let keep it unchange.

File: orttraining/pytorch_frontend_examples/mnist_training.py (hunk 4 above)
Comment (XiaocenDong) [reply orttraining/pytorch_frontend_examples/mnist_training.py:172]: Also can run on single GPU, just use the "python mnist_training.py" command, and use mpi for 2gpu.

**Comment by pengwa** [comment]
/azp run orttraining-linux-gpu-ci-pipeline,orttraining-linux-gpu-ci-pipeline (Onnxruntime_Linux_GPU_Training Debug),orttraining-linux-gpu-ci-pipeline (Onnxruntime_Linux_GPU_Training Release),orttraining-linux-ci-pipeline,orttraining-linux-ci-pipeline (Onnxruntime_Linux_CPU_Training Debug),orttraining-linux-ci-pipeline (Onnxruntime_Linux_CPU_Training Release),orttraining-linux-gpu-inference-only-ci,orttraining-mac-ci-pipeline,orttraining-win-ci-pipeline,orttraining-win-ci-pipeline (Win_CPU_Training Debug),orttraining-win-ci-pipeline (Win_CPU_Training RelWithDebInfo),orttraining-win-gpu-ci-pipeline,orttraining-win-gpu-ci-pipeline (Win_GPU_Training Debug),orttraining-win-gpu-ci-pipeline (Win_GPU_Training RelWithDebInfo)

**Comment by pengwa** [comment]
@SherlockNoMad  @liqunfu, the change is pretty straight forward, I reviewed it and discussed with @XiaocenDong carefully, so will merge it. If you have any concerns, feel free to raise up later.

**Pull request merged**